libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineView.kt

```diff
@@ -142,14 +142,32 @@ class MapboxRouteLineView(options: MapboxRouteLineOptions) {
      * the layers if they have not yet been initialized. If you have a use case for initializing
      * the layers in advance of any API calls this method may be used.
      *
+     * If the layers already exist they will be removed and re-initialized with the options provided.
+     *
      * Each [Layer] added to the map by this class is a persistent layer - it will survive style changes.
      * This means that if the data has not changed, it does not have to be manually redrawn after a style change.
      * See [Style.addPersistentStyleLayer].
      *
      * @param style a valid [Style] instance
      */
     fun initializeLayers(style: Style) {
-        MapboxRouteLineUtils.initializeLayers(style, options)
+        resetLayers(style)
```

Comment:
Let's add the logic that resets layers on first data render call, as we did with the `RouteArrowView`. It's never been a requirement to call `initializeLayers` before using the `RouteLineView` and we shouldn't enforce that now for developers to fix the bug.

Reply:
I believe the implementation will reset the layers on the first render call as is.  I'll test to confirm. If not I propose checking for a change in the options rather than using the flag introduced in `RouteLineView`.

Reply:
> checking for a change in the options

I recall we introduce the flag in the `RouteArrowView` because making a diff between already applied source/layer properties and new ones wasn't possibly to be generated. Unless that's not the case?

Reply:
I was suggesting checking if `MapboxRouteLineOptions` has changed.